src/tpm2-tss-engine-common.c

```diff
@@ -588,3 +588,71 @@ init_tpm_key (ESYS_CONTEXT **esys_ctx, ESYS_TR *keyHandle, TPM2_DATA *tpm2Data)
     esys_ctx_free(esys_ctx);
     return r;
 }
+
+
+/** Initialize the ESYS TPM connection, load a key then evict the key
+ *
+ * Establish a connection with the TPM using ESYS libraries, create a primary
+ * key using the persistent handle, evict the key into persistent storage and
+ * return the result.
+ * @param tpm2Data The key data, owner auth and key auth to be used
+ * @param persKeyHandle The persistent handle of the key
+ * @param owner The handle of the keys owner
+ * @retval TSS2_RC_SUCCESS on success
+ * @retval TSS2_RCs according to the error
+ */
+int tpm2tss_load_and_evict(TPM2_DATA *tpm2Data, ESYS_TR persKeyHandle, ESYS_TR owner)
+{
+    TSS2_RC r = 1;
+    ESYS_TR transKeyHandle = ESYS_TR_NONE;
+    ESYS_TR newKeyHandle = ESYS_TR_NONE;
+    ESYS_TR parentKeyHandle = ESYS_TR_NONE;
+    ESYS_CONTEXT *esys_ctx = NULL;
+
+    if (tpm2Data->privatetype == KEY_TYPE_HANDLE) {
+        fprintf(stderr, "%s: Given tpm2data->privatetype of keyhandle", __func__);
+        goto error;
+    }
+
+    r = init_tpm_key(&esys_ctx, &transKeyHandle, tpm2Data);
+
+    if (r == TSS2_ESYS_RC_MEMORY) {
+        fprintf(stderr, "%s: init Malloc failure\n", __func__);
+        goto error;
+    } else if (r == 0x000009a2) { /* Magic number from tpm2-tss-enigne */
```

Comment:
Not really a magic number from `tpm2-tss-engine`, but the return code from the TPM2 on auth failure.  

This entire if-else block can be replaced by one call to [ERRchktss](https://github.com/tpm2-software/tpm2-tss-engine/blob/368ef845b2f47aced56ee5f25da9ffafb34f1731/src/tpm2-tss-engine-err.h#L58).

Reply:
I originally wrote this to be separate from `tpm2-tss-engine` and didn't have the macro, thus the comment. If we keep this here I'll substitute the macro back in.

Reply:
:+1: